src/scene.js

```diff
@@ -270,9 +270,13 @@ Scene.prototype.setZoom = function (zoom) {
         this.removeTilesOutsideZoomRange(below, above);
     }
 
+
     this.last_zoom = this.zoom;
     this.zoom = zoom;
-    this.capped_zoom = Math.min(Math.round(this.zoom), this.tile_source.max_zoom || Math.round(this.zoom));
+
+    this.capped_zoom = Math.min(Math.round(this.zoom), this.findMaxZoom() || Math.round(this.zoom));
+    this.zooming = false;
```

Comment:
This duplicates the line at the top of the function, and I believe to work correctly there's a state order dependence that requires it to be there before `removeTilesOutsideZoomRange` is called.